packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3599,9 +3600,42 @@ export class TransactionController extends BaseController<
       );
     }
 
+    if (this.#checkIfSimulationRetriggerNeeded(transactionMeta)) {
+      this.#updateSimulationData(transactionMeta).catch((error) => {
+        log('Error updating simulation data', error);
+        throw error;
+      });
+    }
+
     return transactionMeta;
   }
 
+  #checkIfSimulationRetriggerNeeded(transactionMeta: TransactionMeta) {
+    const {
+      securityAlertResponse,
+      simulationData,
+      txParams: { value },
+    } = transactionMeta;
+
+    const isSimulationDataAvailable = Boolean(simulationData);
+    const isMaliciousTransfer =
+      securityAlertResponse?.result_type === BlockaidResultType.Malicious;
+    const simulationNativeBalanceDifference =
+      simulationData?.nativeBalanceChange?.difference;
+    const isBalanceDifferenceEqual =
```

Comment:
Are we definitely looking for an exact comparison rather than a percentage difference?

Reply:
What do we think about the value on this? Would it make sense to make it `5%`?